sale_order_priority/models/sale.py

```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Simone Rubino - Agile Business Group
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import models, fields, api
+from odoo.addons.procurement.models import procurement
+
+
+class SaleOrderLine(models.Model):
+    _inherit = 'sale.order.line'
+
+    priority = fields.Selection(
+        procurement.PROCUREMENT_PRIORITIES, string='Priority', default='1')
+
+    @api.multi
+    def _prepare_order_line_procurement(self, group_id=False):
+        self.ensure_one()
+        res = super(SaleOrderLine, self) \
+            ._prepare_order_line_procurement(group_id=group_id)
+        res['priority'] = self.priority \
+            or self.default_get(['priority'])['priority']
+        return res
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    @api.depends('order_line.priority')
+    def _compute_priority(self):
+        for order in self:
+            order.priority = order.mapped('order_line') and \
+                max(order.mapped('order_line.priority')) or '1'
+
+    def _inverse_priority(self):
+        for order in self:
+            order.order_line.write({'priority': order.priority})
```

Comment:
@SimoRubi You cannot call write in an inverse. Affect recordset values directly

Reply:
@rousseldenis why can't I call `write` in an inverse method? I took it from https://github.com/odoo/odoo/blob/0780f10c5474ae28e768b1bbba0245b0fb46e990/addons/stock/models/stock_picking.py#L341 and tested it, it works.

Moreover I can't `order.order_line = order.priority` because `order.order_line` is `One2Many`

Reply:
@SimoRubi You cannot trust everythong in code but go to https://github.com/odoo/odoo/blob/0780f10c5474ae28e768b1bbba0245b0fb46e990/odoo/fields.py#L206

The problem is that inverse methods are called by write, see https://github.com/odoo/odoo/blob/0780f10c5474ae28e768b1bbba0245b0fb46e990/odoo/models.py#L3586

If you write in inverse, that can trigger some unwanted recomputations.

You have to loop into order_line and affect values.

Reply:
@rousseldenis now I understand what you meant, done, thanks.

Anyway, I checked that assigning directly to the model (https://github.com/odoo/odoo/blob/0780f10c5474ae28e768b1bbba0245b0fb46e990/odoo/models.py#L5238) actually seems to call the `write` method too (https://github.com/odoo/odoo/blob/0780f10c5474ae28e768b1bbba0245b0fb46e990/odoo/fields.py#L925), is it correct or I am mixing up already?